### DISCUSSION TITLE
Request - Search All

### DISCUSSION THREAD

**Discussion by gurukast** [BlafKing/sd-civitai-browser-plus#69] [Ideas]
optional search all, that would search not only model name, but  username and tags as well - or at least model name and tags lol

**Comment by Hamguy21** [comment]
would be nice to be able to select what images to save

**Reply by BlafKing** [reply]
Sounds like a reasonable request, I'll see if I can find a good way to implement this :)

**Comment by BlafKing** [comment]
> optional search all, that would search not only model name, but username and tags as well - or at least model name and tags lol

If I understood you correctly, then this unfortunately isn't supported by [CivitAI's REST API](https://github.com/civitai/civitai/wiki/REST-API-Reference).

For example, if you insert a query for the model name then it shows models with a matching name, however, if you also include a query for either a tag or artist name, then it further filters the search results, so it would only show results that match both the model name and artist name.
Instead of showing results that match either of the query's, it would only show results that match both of the inserted query's.

So currently, it's not possible to make the API Return results that match either inserted query, it only returns results that match both inserted query's.

Providing combined search query's (model name, artist name, tag) is currently not possible with the extension which does sound like a good feature to me, so that users have the ability to further narrow down search results so I'll have a look if I can implement that.

I assume this isn't what you meant though, and with the current API your request isn't possible to implement :/
